lib/rules/sort-vars.js

```diff
@@ -14,9 +15,90 @@ module.exports = function(context) {
     var configuration = context.options[0] || {},
         ignoreCase = configuration.ignoreCase || false;
 
+
+
     return {
         "VariableDeclaration": function(node) {
-            node.declarations.reduce(function(memo, decl) {
+            var nodeDeclarations;
+            var PatternMapper = {};
+
+             // Adds all leaves to single collection
+            PatternMapper.flattenIntoArray = function flattenIntoArray(obj, collection) {
+                if (Array.isArray(obj)) {
```

Comment:
Originally was none object, separate functions that call one another cause a lint error since one of the functions must be defined prior to the other. Probably could be put into one larger function with internal closure, but I wasn't sure how messy that would look. I won't continue working on this if you are planning on just not going to support sorting across destructured variables. Just need something to fix the error.